src/main.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+import { NFC } from 'nfc-pcsc';
+import EthTx from 'ethereumjs-tx';
+import LuksoNfcChip from './reader/LuksoNfcChip';
+import Tlv from './tlv/Tlv';
+
+const nfc = new NFC();
+
+nfc.on('reader', async reader => {
+  console.log(`${reader.reader.name}  device attached`);
+  reader.aid = '654E6F7465734170706C6574';
+  reader.on('card', async card => {
+    console.log();
+
+    // Get Chip Object (Automatic read public key)
+    let chip = await LuksoNfcChip.build(reader);
+    console.log(`Public key:                0x${chip.blkPubKey}`);
+
+    // Transaction Signature Counter
+    let counter = await chip.read(Tlv.TAG_TRANSACTION_SIGNATURE_COUNTER);
+    console.log(`Tx Signature Counter:      0x${counter}`)
+
+    // Challenge-response
+    let result = await chip.verify(Tlv.TAG_PRIVATE_KEY_BLK);
```

Comment:
IWhere is the challenge defined?
Is the TAG_PRIVATE... the challenge?

Reply:
Challenge is a 32byte random number send to chip within APDU command, you can find it at line 81 in LuksoNfcChip.js: 
```
const challenge = crypto.randomBytes(Tlv.LENGTH_CHALLENGE).toString('hex');
```
The input parameter ```TAG_PRIVATE_KEY_BLK``` is used to ask chip to sign challenge with BLOCKCHAIN private key for there are serveral private keys in the chip.